src/h3lib/include/iterators.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file iterators.h
+ * @brief Iterator structs and functions for the children of a cell,
+ * or cells at a given resolution.
+ */
+
+#ifndef ITERATORS_H
+#define ITERATORS_H
+
+#include <stdint.h>
+
+#include "h3api.h"
+
+/*
+Iter_Child: struct for iterating through the descendants of
+a given cell
+ */
+typedef struct {
+    H3Index h;
+    int parentRes;  // parent resolution
+    int fnz;        // first nonzero digit (this digit skips `1` for pentagons)
+} Iter_Child;
+
+DECLSPEC Iter_Child iterInitParent(H3Index h, int childRes);
+DECLSPEC Iter_Child iterInitBaseCellNum(int baseCellNum, int childRes);
+DECLSPEC void iterStepChild(Iter_Child* iter);
```

Comment:
The step functions seem like a good place to have a return code indicating whether there is another cell to evaluate, or perhaps returning the cell itself.

Reply:
I think you have all that functionality with the current `it.h` interface, and it maintains a "there's only one way to do it" quality of the interface.

Also, returning the cell itself is a bit weird, as if you return the *next* cell, then you have to have a special case for the first cell in the iteration. If you return the *previous* cell, I think that's potentially confusing, as that probably isn't what most folks are expecting.